doc/source/contributing.rst

```diff
@@ -103,167 +108,164 @@ want to clone your fork to your machine::
+- https://blog.ionelmc.ro/2014/12/21/compiling-python-extensions-on-windows/
+- https://support.enthought.com/hc/en-us/articles/204469260-Building-Python-extensions-with-Canopy
 
-This will replay your commits on top of the latest pandas git master.  If this
-leads to merge conflicts, you must resolve these before submitting your pull
-request.  If you have uncommitted changes, you will need to ``stash`` them prior
-to updating.  This will effectively store your changes and they can be reapplied
-after updating.
+Let us know if you have any difficulties by `opening an issue
+<https://github.com/pandas-dev/pandas/issues/new>`_.
 
-.. _contributing.dev_env:
+.. _contributiong.dev_python:
 
-Creating a development environment
-----------------------------------
+Creating a Python Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-An easy way to create a *pandas* development environment is as follows.
+Now that you have a C complier, create an isolated pandas development
+environment:
 
-- Install either :ref:`Anaconda <install.anaconda>` or :ref:`miniconda <install.miniconda>`
+- Install either `Anaconda <https://www.anaconda.com/download/>`_ or `miniconda
+  <https://conda.io/miniconda.html>`_
+- Make sure your conda is up to date (`conda update conda`)
 - Make sure that you have :ref:`cloned the repository <contributing.forking>`
 - ``cd`` to the *pandas* source directory
 
-Tell conda to create a new environment, named ``pandas_dev``, or any other name you would like
-for this environment, by running::
-
-      conda create -n pandas_dev --file ci/requirements_dev.txt
-
+We'll now kick off a three-step process:
 
-For a python 3 environment::
-
-      conda create -n pandas_dev python=3 --file ci/requirements_dev.txt
-
-.. warning::
-
-   If you are on Windows, see :ref:`here for a fully compliant Windows environment <contributing.windows>`.
-
-This will create the new environment, and not touch any of your existing environments,
-nor any existing python installation. It will install all of the basic dependencies of
-*pandas*, as well as the development and testing tools. If you would like to install
-other dependencies, you can install them as follows::
+1. Install the build dependencies
+2. Build and install pandas
+3. Install the optional dependencies
 
-      conda install -n pandas_dev -c pandas pytables scipy
+.. code-block:: none
 
-To install *all* pandas dependencies you can do the following::
+   # Create and activate the build environment
+   conda env create -f ci/environment-dev.yaml
+   conda activate pandas-dev
 
-      conda install -n pandas_dev -c conda-forge --file ci/requirements_all.txt
+   # Build and install pandas
+   python setup.py build_ext --inplace -j 4
+   python -m pip install -e .
 
-To work in this environment, Windows users should ``activate`` it as follows::
+   # Install the rest of the optional dependencies
+   conda install -c defaults -c conda-forge --file=ci/requirements-optional-conda.txt
```

Comment:
What does this exactly do by specifying both? First look at defaults and otherwise look at conda-forge?

Reply:
Correct. IIRC we have deps that are conda-forge only. I suppose we maybe just to `-c conda-forge` and no defaults.